src/init.cpp

```diff
@@ -644,6 +645,17 @@ pj_init_ctx_with_allow_init_epsg(PJ_CONTEXT *ctx, int argc, char **argv, int all
         return pj_default_destructor (PIN, PROJ_ERR_INVALID_OP_ILLEGAL_ARG_VALUE);
     }
 
+    /* S2 projection parameter */
```

Comment:
parsing this parameter should be done in the setup function of the projection, not in this generic code

Reply:
@rouault - I've been working with @marcus-elia on this. When you say the setup function of the projection, are you referring to the block of code beginning here?
https://github.com/OSGeo/PROJ/blob/a1b640a8f019ebebf2112d26f1c1ada39ef49194/src/projections/s2.cpp#L310-L314
If so, I'm wondering if you have another example of a projection that parses parameters outside of `src/init.cpp`, as it doesn't appear that `ctx` gets passed through to that code in order to call `pj_param` properly.

Reply:
> When you say the setup function of the projection, are you referring to the block of code beginning here?

yes

> If so, I'm wondering if you have another example of a projection that parses parameters outside of src/init.cpp, as it doesn't appear that ctx gets passed through to that code in order to call pj_param properly.

It is available in P->ctx. See e.g. https://github.com/OSGeo/PROJ/blob/master/src/projections/lcc.cpp#L89

Reply:
Perfect, thanks!